### ISSUE TITLE
Improper translation (zh-cn)

### ISSUE THREAD

**Issue by zms9110750** [dotnet/docs#29724]
Finalizers (historically referred to as destructors) 
Destructors translated as "destructor function" in Chinese

In the sense, it should be translated as "析构方法".(destructors method)
Or the literal translation should be "析构器"(There are no methods and no functions)

In the current document, the "Finalizers" translation has no function or method
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a96eaef8-dcc9-4659-3f4e-be5a1c4c6b50
* Version Independent ID: d3da7981-542d-e906-5988-b80e403fa651
* Content: [Finalizers - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/finalizers)
* Content Source: [docs/csharp/programming-guide/classes-and-structs/finalizers.md](https://github.com/dotnet/docs/blob/main/docs/csharp/programming-guide/classes-and-structs/finalizers.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by GeeLaw** [comment]
The finalizer is different from the destructor and should have never been referred to as a destructor. Since a different term is used in one language (English, which is the source language of the documentation), it’s better to use a different term in every language to avoid confusion.

See https://docs.microsoft.com/en-us/cpp/dotnet/how-to-define-and-consume-classes-and-structs-cpp-cli?view=msvc-170#BKMK_Destructors_and_finalizers

**Comment by zms9110750** [comment]
@GeeLaw 
>it’s better to use a different term in every language to avoid confusion.
Sorry, I forgot to add a reference. Let me rephrase the problem

>Finalizers (historically referred to as destructors)
This sentence is in the document

I mean the word "destructors"
I think it is a dictionaries created by you and translated by machine.
In the dictionary, "destructors" are translated as "destructors function"
This word should be translated as "析构器"

Although the concept is no longer used, the word is only used in a few places. But I still think it should be unified

**Comment by GeeLaw** [comment]
Got your point. You're suggesting that

> Finalizers (historically referred to as **destructors**) are used to perform any necessary final clean-up when a class instance is being collected by the garbage collector.
>
> 终结器（以前称为析构函数）用于在垃圾回收器收集类实例时执行任何必要的最终清理操作。

should instead read

> Finalizers (historically referred to as **destructors**) are used to perform any necessary final clean-up when a class instance is being collected by the garbage collector.
>
> 终结器（以前称为**析构器**）用于在垃圾回收器收集类实例时执行任何必要的最终清理操作。

Right?

**Comment by zms9110750** [comment]
@GeeLaw 
 oh yes.And constructors have the same translation problems.#29725

**Comment by gewarren** [comment]
Ping @kgliang

**Comment by v-alje** [comment]
CEINTL work item 806166 filed to the LOC team on this issue.

**Comment by BillWagner** [comment]
This issue has been imported to the global localization team here:
[Associated WorkItem - 538547](https://dev.azure.com/msft-skilling/Content/_workitems/edit/538547)

Closing this issue because it's been imported into the global localization system. The fix is not in this repository.
